Added documentation for webhooks and document processing tutorials

vibehavjha opened the pull request:
## Category

- [ ] Content fix
- [x] New article

## Related issues

- fixes #NA
- partially #NA
- mentioned in #NA

## What's in this Pull Request?
Added new Using Webhooks and Enable Document Processing with ACS tutorials.

vibehavjha commented:
Hi @marcwindle, please review this PR

andrewconnell commented on docs/embedded/tutorials/doc-processing-acs.md:
```diff
@@ -0,0 +1,165 @@
+![document processing schema](../images/Document-Processing.png)
+
+> [!TIP]
+> To learn more about the Graph APIs used in this tutorial, see [Track changes for a Drive](https://onedrive.visualstudio.com/OneDrive%20Service/_git/apidocs?path=/docs/rest-api/api/driveitem_delta.md), [Get a DriveItem resource](https://onedrive.visualstudio.com/OneDrive%20Service/_git/apidocs?path=/docs/rest-api/api/driveitem_get.md), and [Upload or replace the contents of a DriveItem](https:// ...
```
None of these links work. Must fix

andrewconnell commented on docs/embedded/tutorials/doc-processing-acs.md:
```diff
@@ -0,0 +1,165 @@
+---
+title: Document Processing with ACS
```
Spell out ACS everywhere... it's not commonly used for Azure Cognitive Services, but also easily confused with other services (including the retired Azure Access Control Service).

andrewconnell commented on docs/embedded/tutorials/doc-processing-acs.md:
```diff
@@ -0,0 +1,165 @@
+
+# Enabling Document Processing with ACS
+
+## Utilizing Azure Cognitive Services
```
Proper markdown formatting is to include a line break between each heading & paragraph. See existing docs for reference.

andrewconnell commented on docs/embedded/tutorials/doc-processing-acs.md:
```diff
@@ -0,0 +1,165 @@
+
+
+## Get the delta changes of a container
+Open `GraphProvider.ts` and implement method `getDriveChanges` to get the list of changed items.
```
Filenames should be **bold**, not inline code formatted.

andrewconnell commented on docs/embedded/tutorials/using-webhooks.md:
```diff
@@ -0,0 +1,103 @@
+Webhooks are automated messages that are transmitted by an application when a trigger is activated. They can be used in SPE to enable the automation of workflows, the integration of systems, and to respond to events in real time.
+You will use webhooks to invoke the Azure Cognitive Service APIs from the application whenever an existing file is updated, or a new file is uploaded.
+To set up webhooks with your [current SharePoint Embedded application](/training/modules/sharepoint-embedded-create-app/), you need to
+1.	Create and register a webhook endpoint to get notifications whenever there is change in your container. This will be done using REST APIs.
```
Don't explicitly number bullets. All bullets should be `1.` and let the rendering engine create the correct numbers.

andrewconnell commented:
Thanks for the new doc contribution! There are multiple things we need to address before we can fully review this. Please see my comments above. Feel free to ask questions if anything isn't clear.

vibehavjha commented:
Hi @andrewconnell, I have implemented the suggested changes for both tutorials.

marcwindle commented:
[like]  Marc Windle reacted to your message:
________________________________
From: Vaibhav Jha ***@***.***>
Sent: Tuesday, March 12, 2024 9:01:57 AM
To: SharePoint/sp-dev-docs ***@***.***>
Cc: Marc Windle ***@***.***>; Mention ***@***.***>
Subject: Re: [SharePoint/sp-dev-docs] Added documentation for webhooks and document processing tutorials (PR #9543)


Hi @andrewconnell<https://github.com/andrewconnell>, I have implemented the suggested changes for both tutorials.

—
Reply to this email directly, view it on GitHub<https://github.com/SharePoint/sp-dev-docs/pull/9543#issuecomment-1991084838>, or unsubscribe<https://github.com/notifications/unsubscribe-auth/AZ22CBW3CZJWUMA5WKQQCCDYX3AALAVCNFSM6AAAAABDZZKLPKVHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMYTSOJRGA4DIOBTHA>.
You are receiving this because you were mentioned.Message ID: ***@***.***>

andrewconnell commented:
@vibehavjha said:
> I have implemented the suggested changes for both tutorials.

Can you review & address the automated build issue findings reported in the last build? https://github.com/SharePoint/sp-dev-docs/pull/9543#issuecomment-1991084063

Most of these are suggestions, but MSDOCs does require us to use relative links for learn content, and to not include locale's in the URLs, rather letting the MSFT gateway redirect to the appropriate locale. Also, there's a broken link reference.

andrewconnell reviewed (changes requested):
Please fix absolute & broken links in the automated build report.

vibehavjha commented:
> Please fix absolute & broken links in the automated build report.

I have fixed the absolute links and fixed locales. The broken links in each of the tutorials point to the other one. Please let me know if I should remove the hyperlinks and then add them back once the pages are live.

andrewconnell commented:
I made some grammatical fixes and fixed those links. When you use the leading `/file.md`, that says "go back to the root of the repo" which isn't correct and will create broken links. To stay in the same folder, you use `./file.md` which resolves both of the broken link issues.

vibehavjha commented:
Thank you @andrewconnell, I'll remember the suggested changes for future articles.
